forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -241,3 +278,191 @@ impl WalletCommands {
+            let mut fil = num
+                .round_sf_with_strategy(
+                    NUM_SIGNIFICANT_DIGITS,
+                    RoundingStrategy::MidpointAwayFromZero,
+                )
+                .ok_or(anyhow::Error::msg("cannot represent"))?;
+            if fil == fil.trunc() {
+                fil = fil.trunc();
+            }
+            let mut res = format!("{} {} FIL", fil, units[unit_index]);
+
+            if fil != num {
+                res.insert(0, '~');
+            }
+
+            res
+        }
+    };
+    Ok(res)
+}
+
+fn bool_pair_to_mode(exact: bool, fixed: bool) -> FormattingMode {
+    if exact && fixed {
+        FormattingMode::ExactFixed
+    } else if !exact && fixed {
+        FormattingMode::NotExactFixed
+    } else if exact && !fixed {
+        FormattingMode::ExactNotFixed
+    } else {
+        FormattingMode::NotExactNotFixed
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use fvm_shared::econ::TokenAmount;
+
+    use super::*;
+
+    #[test]
+    fn exact_balance_fixed_unit() {
+        let cases_vec = vec![
+            (100, "0.0000000000000001 FIL"),
+            (12465, "0.000000000000012465 FIL"),
+            (1508900000000005000, "1.508900000000005 FIL"),
+        ];
+
+        for (atto, result) in cases_vec {
+            test_call(atto, result, true, true);
+        }
+    }
+
+    #[test]
+    fn not_exact_balance_fixed_unit() {
+        let cases_vec = vec![
+            (100, "0.0000000000000001000 FIL"),
+            (999999999999999999, "~1.0000 FIL"),
+            (1000005000, "~0.000000001000 FIL"),
+            (508900000000005000, "~0.5089 FIL"),
+            (1508900000000005000, "~1.509 FIL"),
+            (2508900009000005000, "~2.509 FIL"),
+        ];
+
+        for (atto, result) in cases_vec {
+            test_call(atto, result, false, true);
+        }
+    }
+
+    #[test]
+    fn exact_balance_not_fixed_unit() {
+        let cases_vec = vec![
+            (100, "100 atto FIL"),
+            (120005, "120.005 femto FIL"),
+            (200000045, "200.000045 pico FIL"),
+            (1000000123, "1.000000123 nano FIL"),
+            (450000008000000, "450.000008 micro FIL"),
+            (90000002750000000, "90.00000275 milli FIL"),
+            (1508900000000005000, "1.508900000000005  FIL"),
+            (2508900009000005000, "2.508900009000005  FIL"),
+        ];
+
+        for (atto, result) in cases_vec {
+            test_call(atto, result, true, false);
+        }
+    }
+
+    #[test]
+    fn not_exact_balance_not_fixed_unit() {
+        let cases_vec = vec![
+            (100, "100 atto FIL"),
+            (120005, "~120 femto FIL"),
+            (200000045, "~200 pico FIL"),
+            (1000000123, "~1 nano FIL"),
+            (450000008000000, "~450 micro FIL"),
+            (90000002750000000, "~90 milli FIL"),
+            (500236779800000000, "~500.2 milli FIL"),
+            (1508900000000005000, "~1.509  FIL"),
```

Comment:
I don't think we want two spaces between the amount and the unit.

```suggestion
            (1508900000000005000, "~1.509 FIL"),
```

Reply:
Fixed